lisa/tests/hotplug.py

```diff
@@ -269,4 +271,174 @@ def cpuhp_seq(cls, nr_operations, hotpluggable_cpus, max_cpus_off, random_gen):
         for cpu in cur_off_cpus:
             yield cpu, 1
 
-# vim :set tabstop=4 shiftwidth=4 textwidth=80 expandtab
+
+class HotplugRollback(FtraceTestBundle):
+
+    @classmethod
+    def _online(cls, target, cpu, online, verify=True):
+        try:
```

Comment:
As a general rule the try/except should only include what we strictly want to check:

```@classmethod
def _online(cls, target, cpu, online, verify=True):
    hp = target.hotplug
    f = hp.online if online else hp.offline
    try:
        f(cpu)
    except Exception:
        if verify:
            raise
```

`if foo: ...` is equivalent to `if bool(foo): ...`, itself equivalent (at least for all classes defined in Python) to `if foo.__bool__(): ...`: 
```
>>> class C:
...  def __bool__(self):
...   raise ValueError()
... 
>>> c=C()
>>> if c: pass
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 3, in __bool__
ValueError
```
Other example:
```
>>> import pandas as pd
>>> df=pd.DataFrame()
>>> if df: pass
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.9/site-packages/pandas/core/generic.py", line 1442, in __nonzero__
    raise ValueError(
ValueError: The truth value of a DataFrame is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

So there is no guarantee that even a check like that won't raise an exception. The problem is made worse by the fact that `Exception` is caught, which is probably way too wide (I suppose we actually expect a TargetStableError).